### PR TITLE
Add docker-py stubs

### PR DESCRIPTION
**Pull request by kasium**

### CODE REVIEW

**Comment by kasium** [comment]
So the mypy_primer errors seem justified. Some invalid imports (they work at runtime due to side-effects but are not allowed). In addition in `perfect`, they seem to use decorators making an async method sync which is not correctly typed leading now to errors when DockerClient is no longer Any

**Comment by Viicos** [comment]
There is already some precedent here: https://github.com/docker/docker-py/issues/2796

**Comment by kasium** [comment]
> There is already some precedent here: [docker/docker-py#2796](https://github.com/docker/docker-py/issues/2796)

I'm aware of this issue. There was no real progress since 3 years...

**Comment by srittau** [comment]
Stubtest still has a few complaints.

File: stubs/docker/docker/constants.pyi [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+INSECURE_REGISTRY_DEPRECATION_WARNING: str
+IS_WINDOWS_PLATFORM: bool
+WINDOWS_LONGPATH_PREFIX: str
+DEFAULT_USER_AGENT: str
+DEFAULT_NUM_POOLS: int
+DEFAULT_NUM_POOLS_SSH: int
+DEFAULT_MAX_POOL_SIZE: int
+DEFAULT_DATA_CHUNK_SIZE: int
+DEFAULT_SWARM_ADDR_POOL: Sequence[str]
+DEFAULT_SWARM_SUBNET_SIZE: int
```
Comment (srittau) [inline stubs/docker/docker/constants.pyi:21]: Optional: Considering this module is called "constants", would it make sense to annotate these using `Final`?

**Review by srittau — APPROVED** [review]
LGTM after a casual scan. One optional suggestion below.

File: stubs/docker/docker/constants.pyi (hunk 1 above)
Comment (kasium) [reply stubs/docker/docker/constants.pyi]: done

**Comment by adamtheturtle** [comment]
Thank you @kasium and @srittau !

**Pull request merged**